### PR TITLE
MAINT: mypy: fix remaining type errors

### PR DESCRIPTION
**Pull request by person142**
#### Reference issue
None

#### What does this implement/fix?
This fixes the remaining mypy errors. One error appears to be an issue in the NumPy stubs:

https://github.com/numpy/numpy-stubs/issues/42

The other is fixed by modifying how we import `linalg_version` inside the linalg setup.

#### Additional information
None

### CODE REVIEW

File: scipy/linalg/setup.py [hunk 1]
```diff
@@ -118,7 +118,7 @@ def configuration(parent_package='', top_path=None):
 
 if __name__ == '__main__':
     from numpy.distutils.core import setup
-    from linalg_version import linalg_version
+    from scipy.linalg.linalg_version import linalg_version
```
Comment (person142) [inline scipy/linalg/setup.py]: I think this is equivalent? (And, if so, seems like a better practice?)

Comment (ilayn) [reply scipy/linalg/setup.py]: Yes this relative import business is a mess actually. See this post for more info https://stackoverflow.com/a/28154841

**Comment by rgommers** [comment]
Just delete `linalg_version` completely (in `setup.py` and `__init__.py`) I'd say, it's a relic that's not in our API docs and is complete useless (version 0.4.9 of what?).

File: scipy/io/matlab/tests/test_mio.py [hunk 2]
```diff
@@ -137,7 +137,12 @@ def mlarr(*args, **kwargs):
      })
 st_sub_arr = array([np.sqrt(2),np.exp(1),np.pi]).reshape(1,3)
 dtype = [(n, object) for n in ['stringfield', 'doublefield', 'complexfield']]
-st1 = np.zeros((1,1), dtype)
+# See
+#
+# https://github.com/numpy/numpy-stubs/issues/42
+#
+# for the reasoning behind the ignore.
```
Comment (rgommers) [inline scipy/io/matlab/tests/test_mio.py]: out of curiosity: why on 5 lines rather than 1 or 2?

**Comment by person142** [comment]
> Just delete linalg_version completely

Escaping through the horns of the dilemma-deleted it.

**Comment by rgommers** [comment]
LGTM, merged. Thanks @person142

**Pull request merged**